LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/OTAAHelper/OTAAKeysGenerator.cs

```diff
@@ -10,19 +10,15 @@ namespace LoRaTools
 
     public static class OTAAKeysGenerator
     {
-        private static readonly RandomNumberGenerator RndKeysGenerator = RandomNumberGenerator.Create();
-
-        public static string GetNwkId(byte[] netId)
+        public static DevAddr GetNwkId(byte[] netId)
         {
             if (netId is null) throw new ArgumentNullException(nameof(netId));
 
-            var nwkPart = netId[0] << 1;
-            var devAddr = new byte[4];
-
-            RndKeysGenerator.GetBytes(devAddr);
-
-            devAddr[0] = (byte)((nwkPart & 0b11111110) | (devAddr[0] & 0b00000001));
-            return ConversionHelper.ByteArrayToString(devAddr);
+            Span<byte> bytes = stackalloc byte[4];
+            RandomNumberGenerator.Fill(bytes);
+            var id = netId[0] & 0b0111_1111;
```

Comment:
Don't we have to shift the `netId` by 1 left here?

Reply:
That's being done by `DevAddr` when the network ID gets passed to it.